tests/pytests/unit/states/test_jboss7.py

```diff
@@ -0,0 +1,589 @@
+import pytest
+import salt.states.jboss7 as jboss7
+from salt.exceptions import CommandExecutionError
+from tests.support.mock import MagicMock, patch
+
+
+@pytest.fixture
+def configure_loader_modules():
+    return {
+        jboss7: {
+            "__salt__": {
+                "jboss7.read_datasource": MagicMock(),
```

Comment:
**suggestion(non-blocking)**: `create_autospec` is preferred, and could be used here with:

```
import salt.modules.jboss7 as jboss_mod
from tests.support.mock import create_autospec

...

"jboss7.read_datasource": create_autospec(jboss_mod),
...
```

Not sure if autospec can happen for an entire module, but that could be another option.